### PR TITLE
Cocoapods support for audit

### PR DESCRIPTION
**Pull request by barv-jfrog**
- [ ] The pull request is targeting the `dev` branch.
- [ ] The code has been validated to compile successfully by running `go vet ./...`.
- [ ] The code has been formatted properly using `go fmt ./...`.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-cli-security/actions/workflows/analysis.yml) passed.
- [ ] All [tests](https://github.com/jfrog/jfrog-cli-security/actions/workflows/test.yml) have passed. If this feature is not already covered by the tests, new tests have been added.
- [ ] All changes are detailed at the description. if not already covered at [JFrog Documentation](https://github.com/jfrog/documentation), new documentation have been added.

-----
Description:
jf audit for cocoapods package manager

-----
Depends on:
https://github.com/jfrog/jfrog-cli-core/pull/1274

-----
audit example on nanopb package in a cocoapods project:
image

-----
Minimum pod version - 1.15.2

### CODE REVIEW

File: commands/audit/sca/cocoapods/cocoapods.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+							snippet += "\n" + lines[snippetLine][startLine:]
+						case endLine - startLine:
+							snippet += "\n" + lines[snippetLine][:endCol]
+						default:
+							snippet += "\n" + lines[snippetLine]
+						}
+					}
+				}
+				podPositions = append(podPositions, sarifutils.CreateLocation(descriptorPath, startLine, endLine, startCol, endCol, snippet))
+				foundDependency = false
```
Comment (attiasas) [inline commands/audit/sca/cocoapods/cocoapods.go:76]: extract this code to its own method for readability and testing --> parsePodfileLineForDependencyLocations(...) []*sarifLocation
try to also extract the inner code more, for example --> createDependencyLocation(...) *sarifLocation

File: commands/audit/sca/cocoapods/cocoapods.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+			if i > tempIndex && foundDependency && strings.Contains(line, "pod") {
+				foundDependency = false
+			} else if foundDependency && strings.Contains(line, directDependencyVersion) {
+				endLine = i
+				endCol = len(line)
+				var snippet string
+				if endLine == startLine {
+					snippet = lines[startLine][startCol:endCol]
+				} else {
+					for snippetLine := 1; snippetLine < endLine-startLine+1; snippetLine++ {
```
Comment (attiasas) [inline commands/audit/sca/cocoapods/cocoapods.go:64]: add a comment on this piece of code or extract it to its own method.
why snippetLine starts at 1?

File: commands/audit/sca/cocoapods/cocoapods.go [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+
+func extractPodsSection(filePath string) (string, error) {
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		return "", err
+	}
+	content := string(data)
+	startIndex := strings.Index(content, "PODS:")
+	if startIndex == -1 {
+		return "", fmt.Errorf("PODS: section not found")
```
Comment (attiasas) [inline commands/audit/sca/cocoapods/cocoapods.go:168]: is `PODS` section required for all cocoapods projects? if it is possiable to have a project like it, we will alway return error on the project with no dependencies

**Review by attiasas — CHANGES_REQUESTED** [review]
Nice, take a look at my comments:
1. Add integration tests to `artifactory_test.go`
2. Add integraion tests to `audit_test.go`
3. Add screen shot of the table results running on a cocoapods project to the PR details

File: utils/techutils/techutils.go [hunk 4]
```diff
@@ -46,27 +47,29 @@ var AllTechnologiesStrings = []string{Maven.String(), Gradle.String(), Npm.Strin
...
-	Java       CodeLanguage = "java"
-	CSharp     CodeLanguage = "C#"
-	CPP        CodeLanguage = "C++"
+	JavaScript    CodeLanguage = "javascript"
+	Python        CodeLanguage = "python"
+	GoLang        CodeLanguage = "go"
+	Java          CodeLanguage = "java"
+	CSharp        CodeLanguage = "C#"
+	CPP           CodeLanguage = "C++"
+	CocoapodsLang CodeLanguage = "any"
```
[the remark this answers is not in the document]
Comment (srmish-jfrog) [reply utils/techutils/techutils.go:56]: Cocoapods actually houses packages built with many languages, not just ObjC / Swift

File: tests/testdata/projects/package-managers/go/missing-context/go.sum [hunk 5]
```diff
@@ -1 +1,542 @@
-github.com/hashicorp/consul v1.9.1/go.mod h1:RQlaP4r7KdNLaPDuihkvghhdvZVOuVlUhlz7HtvC1UI=
\ No newline at end of file
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
[the remark this answers is not in the document]
Comment (srmish-jfrog) [reply tests/testdata/projects/package-managers/go/missing-context/go.sum:1]: Will be removed from PR

File: commands/audit/sca/cocoapods/cocoapods.go [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+		endIndex = strings.Index(subContent, "SPEC REPOS:")
+	}
+	if endIndex != -1 {
+		subContent = subContent[:endIndex]
+	}
+	return subContent, nil
+}
+
+func GetDependenciesData(exePath, currentDir string) (string, error) {
+	_, _, err := cocoapods.RunPodCmd(exePath, currentDir, []string{"install"})
```
[the remark this answers is not in the document]
Comment (srmish-jfrog) [reply commands/audit/sca/cocoapods/cocoapods.go:182]: Will add a test to skip installation if `Podfile.lock` exists and has newer ModTime than `Podfile`

File: commands/audit/sca/cocoapods/cocoapods.go [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+	"github.com/owenrumney/go-sarif/v2/sarif"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+	VersionForMainModule = "0.0.0"
```
[the remark this answers is not in the document]
Comment (barv-jfrog) [reply commands/audit/sca/cocoapods/cocoapods.go:22]: we don't have information on the current package. only on its dependencies, I will add a comment.

File: commands/audit/sca/cocoapods/cocoapods_test.go [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+	assert.Len(t, rootNode[0].Nodes, 4)
+
+	child1 := tests.GetAndAssertNode(t, rootNode[0].Nodes, "GTMSessionFetcher:2.3.0")
+	assert.Len(t, child1.Nodes, 0)
+
+	child2 := tests.GetAndAssertNode(t, rootNode[0].Nodes, "GoogleSignIn:6.2.4")
+	assert.Len(t, child2.Nodes, 2)
+}
+
+func TestGetTechDependencyLocation(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (barv-jfrog) [reply commands/audit/sca/cocoapods/cocoapods_test.go:60]: added

File: commands/audit/sca/cocoapods/cocoapods.go (hunk 3 above)
Comment (barv-jfrog) [reply commands/audit/sca/cocoapods/cocoapods.go:168]: yes its part of the Podfile.lock syntax

File: commands/audit/sca/cocoapods/cocoapods_test.go [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+	_, cleanUp := sca.CreateTestWorkspace(t, filepath.Join("projects", "package-managers", "cocoapods"))
+	defer cleanUp()
+	currentDir, err := coreutils.GetWorkingDirectory()
+	assert.NoError(t, err)
+	locations, err := GetTechDependencyLocation("GoogleSignIn", "6.2.4", filepath.Join(currentDir, "Podfile"))
+	assert.NoError(t, err)
+	assert.Len(t, locations, 1)
+	assert.Equal(t, *locations[0].PhysicalLocation.Region.Snippet.Text, "GoogleSignIn', '~> 6.2.4'")
```
[the remark this answers is not in the document]
Comment (barv-jfrog) [reply commands/audit/sca/cocoapods/cocoapods_test.go:68]: Ok

File: commands/audit/sca/cocoapods/cocoapods.go (hunk 1 above)
Comment (barv-jfrog) [reply commands/audit/sca/cocoapods/cocoapods.go:76]: I do prefer extracting only the outer code, I think overextracting is overcomplicating, what do you think?

File: utils/techutils/techutils.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (attiasas) [reply utils/techutils/techutils.go:56]: please add this as a comment above.
Change the name of the const to `Any` / `Multiple`

**Review by attiasas — CHANGES_REQUESTED** [review]
please add one integration test for audit on a cocoapods project.

Add more details at the PR description for the user, like: `Minimum supported Pod version`...

File: utils/techutils/techutils.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (barv-jfrog) [reply utils/techutils/techutils.go:56]: ok

**Review by attiasas — APPROVED** [review]

**Pull request merged**